File: segundos 4 lenguajes/c#/ProyectoCsharp/JugadorAtaque.cs
namespace ProyectoJugadores
{
    public class JugadorAtaque
    {
        public string Nombre { get; set; }
        public int Nivel { get; set; }

        public JugadorAtaque()
        {
            Nombre = "";
            Nivel = 0;
        }

        public JugadorAtaque(string nombre, int nivel)
        {
            Nombre = nombre;
            Nivel = nivel;
        }

        public static JugadorAtaque Copy(JugadorAtaque j)
        {
            return new JugadorAtaque(j.Nombre, j.Nivel);
        }
    }
}

File: segundos 4 lenguajes/go/JugadorAtaque.go
package main

type JugadorAtaque struct {
	Nombre string
	Nivel  int
}

func NewJugadorAtaque() JugadorAtaque {
	return JugadorAtaque{"", 0}
}

func NewJugadorAtaqueParams(nombre string, nivel int) JugadorAtaque {
	return JugadorAtaque{nombre, nivel}
}

func NewJugadorAtaqueNivel(nivel int) JugadorAtaque {
	return JugadorAtaque{"Anon", nivel}
}

func NewJugadorAtaqueCopy(j JugadorAtaque) JugadorAtaque {
	return JugadorAtaque{j.Nombre, j.Nivel}
}

type AbstractAtaque interface {
	Golpear()
}
